fv3core/utils/global_config.py

```diff
@@ -99,25 +107,53 @@ def __eq__(self, other):
         except AttributeError:
             return False
 
+    def _get_backend_opts(
+        self,
+        func: Optional[Callable] = None,
+        device_sync: Optional[bool] = None,
+        format_source: Optional[bool] = None,
+    ) -> Dict[str, Any]:
+        if not StencilConfig._all_backend_opts:
+            StencilConfig._all_backend_opts = read_backend_options_file()
+        all_backend_opts: Dict[str, Any] = StencilConfig._all_backend_opts
+
+        candidate_opts: Dict[str, Any] = {}
+        if "all" in all_backend_opts:
+            candidate_opts.update(all_backend_opts["all"])
+
+        if func is not None:
+            stencil_name = f"{func.__module__.split('.')[-1]}.{func.__name__}"
+            if stencil_name in all_backend_opts:
+                candidate_opts.update(all_backend_opts[stencil_name])
+
+        backend_opts: Dict[str, Any] = {}
+        for name, option in candidate_opts.items():
+            if "backend" not in option or re.match(option["backend"], get_backend()):
+                backend_opts[name] = option["value"]
```

Comment:
The way this is currently written, you can only set the target value for a single backend. In other words, `name` here is unique across entries. Is this sufficient, or is it worth restructuring the yaml file to use list entries allowing configuration of the same name on different backends?

I would also suggest making a named variable e.g. `using_option_backend = re.match(pattern=option.get("backend", ""), string=get_backend())` then `if using_option_backend:` to make it a little clearer what this if statement is triggering on. I had to read it a few times before figuring it out.